### PR TITLE
perf(arm support): add progress bar when deploying arm templates

### PR DESCRIPTION
**Pull request by dilin-MS**

### CODE REVIEW

File: packages/fx-core/src/plugins/solution/fx-solution/constants.ts [hunk 1]
```diff
@@ -168,3 +169,17 @@ export enum SolutionTelemetrySuccess {
 }
 
 export const SolutionTelemetryComponentName = "solution";
+
+export class Messages {
+  // Progress bar messages
+  static readonly DeployArmTemplatesProgressTitle = "Deploying Arm Templates";
```
Comment (jayzhang) [inline packages/fx-core/src/plugins/solution/fx-solution/constants.ts:175]: Suggest: it's better to place constant strings in `fx-core/resource/strings.json` file

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts [hunk 2]
```diff
@@ -143,7 +150,7 @@ export async function deployArmTemplates(ctx: SolutionContext): Promise<Result<v
   );
   const armTemplateJsonFilePath = path.join(azureInfraDir, templateFolder, armTemplateJsonFileName);
   await compileBicepToJson(orchestrationFilePath, armTemplateJsonFilePath);
-  ctx.logProvider?.info("[Solution] Successfully compile bicep files to JSON.");
+  ctx.logProvider?.info("Successfully compile bicep files to JSON.");
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/solution/fx-solution/arm.ts:153]: Why we removed the prefix?

File: packages/fx-core/src/plugins/solution/fx-solution/utils/progressHelper.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { IProgressHandler, PluginContext } from "../../../../../../api/build";
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/solution/fx-solution/utils/progressHelper.ts:1]: maybe we need to reference from teamsfx-api package?

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts (hunk 2 above)
Comment (dilin-MS) [reply packages/fx-core/src/plugins/solution/fx-solution/arm.ts:153]: Only the start and the end message will prefix "solution". Or do we have a consistent practice? @1yefuwang1  @jayzhang

File: packages/fx-core/src/plugins/solution/fx-solution/utils/progressHelper.ts (hunk 3 above)
Comment (dilin-MS) [reply packages/fx-core/src/plugins/solution/fx-solution/utils/progressHelper.ts:1]: Fixed.

File: packages/fx-core/src/plugins/solution/fx-solution/constants.ts (hunk 1 above)
Comment (dilin-MS) [reply packages/fx-core/src/plugins/solution/fx-solution/constants.ts:175]: Moved.

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts (hunk 2 above)
Comment (dilin-MS) [reply packages/fx-core/src/plugins/solution/fx-solution/arm.ts]: Will add the prefix.

**Review by jayzhang — APPROVED** [review]

**Pull request merged**